lnbits/core/tasks.py

```diff
@@ -1,20 +1,103 @@
 import asyncio
-from typing import Dict
+from typing import Dict, Optional
 
 import httpx
 from loguru import logger
 
+from lnbits.settings import get_wallet_class, settings
 from lnbits.tasks import SseListenersDict, register_invoice_listener
 
 from . import db
-from .crud import get_balance_notify, get_wallet
+from .crud import get_balance_notify, get_total_balance, get_wallet
 from .models import Payment
-from .services import websocketUpdater
+from .services import switch_to_voidwallet, websocketUpdater
 
 api_invoice_listeners: Dict[str, asyncio.Queue] = SseListenersDict(
     "api_invoice_listeners"
 )
 
+killswitch: Optional[asyncio.Task] = None
+watchdog: Optional[asyncio.Task] = None
+
+
+async def register_killswitch():
+    """
+    Registers a killswitch which will check lnbits-status repository
+    for a signal from LNbits and will switch to VoidWallet if the killswitch is triggered.
+    """
+    logger.debug("Starting killswitch task")
+    global killswitch
+    killswitch = asyncio.create_task(killswitch_task())
+
+
+async def unregister_killswitch():
+    """
+    Unregisters a killswitch taskl
+    """
+    global killswitch
+    if killswitch:
+        logger.debug("Stopping killswitch task")
+        killswitch.cancel()
+
+
+async def killswitch_task():
+    while True:
+        if settings.lnbits_killswitch:
+            with httpx.Client() as client:
+                try:
+                    r = client.get(settings.lnbits_status_manifest, timeout=4)
+                    r.raise_for_status()
+                    if r.status_code == 200:
+                        ks = r.json().get("killswitch")
```

Comment:
- having a data type for the response would be helpful in documenting how the `status manifest` data structure is expected too look like

Reply:
on the backend side we only ever need this killswitch variable, i think it not necessary just for that. the rest happens only on the frontend.